File: python/5. Longest Palindromic Substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        longestPal = ""
        lenLongest = -1
        for i in range(n):
             #odd case
            l, r = i, i
            while l >= 0 and r < n and s[l] == s[r]:
                curr_len = r-l+1
                if curr_len > lenLongest:
                    longestPal = s[l:r+1]
                    lenLongest = curr_len
                l -= 1
                r += 1
            #even case
            l, r = i , i+1
            while l >= 0 and r < n and s[l] == s[r]:
                curr_len = r-l+1
                if curr_len > lenLongest:
                    longestPal = s[l:r+1]
                    lenLongest = curr_len
                l -= 1
                r += 1
        return longestPal

File: java/5. Longest Palindromic Substring.java
class Solution {
    public String longestPalindrome(String s) {
        int lenS = s.length();
        int lenLongestPal = -1;
        String longestPal = "";
        for(int i = 0; i < lenS; i++){
            //odd case
            int left = i;
            int right = i;
            while( left >= 0 && right < lenS && s.charAt(left) == s.charAt(right)){
                int curr_len = right - left + 1;
                if(curr_len > lenLongestPal){
                    longestPal = s.substring(left, right + 1);
                    lenLongestPal = curr_len;
                }
                left -= 1;
                right += 1;
            }
            //even case
            left = i;
            right = i+1;
            while(left >= 0 && right < lenS &&  s.charAt(left) == s.charAt(right)){
                int curr_len = right - left + 1;
                if( curr_len > lenLongestPal){
                    longestPal = s.substring(left, right + 1);
                    lenLongestPal = curr_len;
                }
                left -= 1;
                right += 1;
            }
        }
        return longestPal;
        
    }
}